packages/statemanager/README.md

```diff
@@ -62,33 +67,63 @@ Have a loot at the extended `CacheOptions` on how to use and leverage the new ca
 
 ```ts
-// setup `stateManager` with some existing address
-const proof = await stateManager.getProof(address)
-const proofWithStorage = await stateManger.getProof(contractAddress, [storageKey1, storageKey2])
-
-const partialStateManager = await DefaultStateManager.fromProof(proof)
-// To add more proof data, use `addProofData`
-await partialStateManager.addProofData(proofWithStorage)
-const accountFromNewSM = await partialStateManager.getAccount(address)
-const accountFromOldSM = await stateManager.getAccount(address)
-console.log(accountFromNewSM, accountFromOldSM) // should match
-const slot1FromNewSM = await stateManager.getContractStorage(contractAddress, storageKey1)
-const slot2FromNewSM = await stateManager.getContractStorage(contractAddress, storageKey1) // should also match
+// ./examples/fromProofInstantiation.ts
+
+import { Account, Address, bytesToHex } from '@ethereumjs/util'
+import { DefaultStateManager } from '@ethereumjs/statemanager'
+import { hexToBytes } from '@ethereumjs/util'
+
+const main = async () => {
+  // setup `stateManager` with some existing address
+  const stateManager = new DefaultStateManager()
+  const contractAddress = new Address(hexToBytes('0xa94f5374fce5edbc8e2a8697c15331677e6ebf0b'))
+  const byteCode = hexToBytes('0x67ffffffffffffffff600160006000fb')
+  const storageKey1 = hexToBytes(
+    '0x0000000000000000000000000000000000000000000000000000000000000001'
+  )
+  const storageKey2 = hexToBytes(
+    '0x0000000000000000000000000000000000000000000000000000000000000002'
+  )
+
+  await stateManager.putContractCode(contractAddress, byteCode)
+  await stateManager.putContractStorage(contractAddress, storageKey1, hexToBytes('0x01'))
+  await stateManager.putContractStorage(contractAddress, storageKey2, hexToBytes('0x02'))
+
+  const proof = await stateManager.getProof(contractAddress)
+  const proofWithStorage = await stateManager.getProof(contractAddress, [storageKey1, storageKey2])
+
+  const partialStateManager = await DefaultStateManager.fromProof(proof)
+  // To add more proof data, use `addProofData`
+  await partialStateManager.addProofData(proofWithStorage)
+  const accountFromNewSM = await partialStateManager.getAccount(contractAddress)
+  const accountFromOldSM = await stateManager.getAccount(contractAddress)
+  console.log(accountFromNewSM, accountFromOldSM) // should match
+
+  const slot1FromNewSM = await stateManager.getContractStorage(contractAddress, storageKey1)
+  const slot2FromNewSM = await stateManager.getContractStorage(contractAddress, storageKey1)
+  console.log(slot1FromNewSM, slot2FromNewSM) // should also match
+}
+main()
 ```
 
 ### `RPCStateManager`
 
 First, a simple example of usage:
```

Comment:
Is there a ref of what RPCStateManager actually does?

Reply:
(it is unclear: we know that over RPC we now request unknown data, but via this example this is not clear)

Reply:
I agree it's unclear, so I moved the explanation for `RPCStateManager` to before the example.